### PR TITLE
chore: refactor dependencies

### PR DESCRIPTION
**Pull request by lawrenceadams**
This pull request updates the project's dependencies and requirements. It removes unused dependencies from the requirements.txt file and adds common requirements to requirements/common.in - allowing deterministic builds for devcontainers/codespaces/ci - and allowing easier updates of packages.

Compiled output requirement files are generated with `uv pip compile ...`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,7 +36,9 @@ dbt-env\Scripts\activate            # activate the environment for Windows
 ```
  4. In your virtual environment, install dbt and other required dependencies as follows:
 ```bash
-pip3 install -r requirements.txt
+# choose requirements depending on which database you're targeting
+pip3 install -r requirements/duckdb.txt
+pip3 install -r requirements/postgres.txt
 pre-commit install
 ```
    - This will install dbt-core, the dbt duckdb and postgres adapters, SQLFluff (a SQL linter),  pre-commit (in order to run SQLFluff on all newly-committed code in this repo), duckdb (to support bootstrapping scripts), and various dependencies for the listed packages
```
Comment (katy-sadowski) [inline README.md:44]: let's move separate ones down into the duckdb and Postgres setup instructions?

**Review by katy-sadowski — CHANGES_REQUESTED** [review]
I tested locally and it works great. Thanks for the clean up :)

One small change suggested and then good to go.

File: README.md (hunk 1 above)
Comment (lawrenceadams) [reply README.md:44]: Good point - makes more sense! Will do this later!

File: README.md [hunk 2]
```diff
@@ -36,13 +36,21 @@ dbt-env\Scripts\activate            # activate the environment for Windows
...
```
Comment (katy-sadowski) [inline README.md:44]: can remove this now!  and maybe just add the specification that requirements needs to be installed in the virtual env (maybe obvious, but can't be too careful :))

Comment (lawrenceadams) [reply README.md:44]: ahhh dim moment - thought I had!!

**Review by katy-sadowski — APPROVED** [review]
🏇 🏇 🏇 🏇

**Pull request merged**